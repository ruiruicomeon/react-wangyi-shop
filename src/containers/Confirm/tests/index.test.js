import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router';

import history from '@/utils/history';
import configureStore from '@/store';

import { Confirm } from '../index';

describe('Confirm 测试', () => {
  let store;

  beforeAll(() => {
    store = configureStore();
  });

  it('snapshot', () => {
    let checkedCarts = [
      {
        id: 1,
        specs: [1],
        images: ['a', 'b'],
        productSpecs: [
          [
            {
              id: 1,
              order: 0,
              index: 0,
              spec: '规格',
              title: '规格1',
            },
          ],
        ],
        productInfo: {
          prices: [179],
        },
        quantity: 1,
      },
    ];

    const tree = renderer
      .create(
        <Provider store={store}>
          <ConnectedRouter history={history}>
            <Confirm checkedCarts={checkedCarts} />
          </ConnectedRouter>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
