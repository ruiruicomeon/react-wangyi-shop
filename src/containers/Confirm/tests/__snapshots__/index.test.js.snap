// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm 测试 snapshot 1`] = `
<div
  className="confirm"
>
  <nav
    className="nav"
  >
    <div
      className="content container"
    >
      <div
        className="notice"
      >
        <img
          alt="speaker"
          className="notice__img"
          src="speaker.gif"
        />
        <ul
          className="notice__list"
        >
          <li
            className="notice__item"
          />
        </ul>
      </div>
      <div
        className="link"
      >
        <ul
          className="link__list"
        >
          <li
            className="link__item"
          >
            <div
              className="link__link"
              onClick={[Function]}
            >
              登录/注册
            </div>
          </li>
          <li
            className="link__item"
          >
            <div
              className="link__link"
            >
              会员
            </div>
          </li>
          <li
            className="link__item"
          >
            <div
              className="link__link"
            >
              甄选家
            </div>
          </li>
          <li
            className="link__item"
          >
            <div
              className="link__link ant-dropdown-trigger"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              企业采购 
              <span
                aria-label="down"
                className="anticon anticon-down"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="down"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                  />
                </svg>
              </span>
            </div>
          </li>
          <li
            className="link__item"
          >
            <div
              className="link__link ant-dropdown-trigger"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              客户服务 
              <span
                aria-label="down"
                className="anticon anticon-down"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="down"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                  />
                </svg>
              </span>
            </div>
          </li>
          <li
            className="link__item"
          >
            <div
              className="link__link"
            >
              <span
                className="link__appIcon"
              />
              APP
            </div>
          </li>
        </ul>
      </div>
    </div>
    <div
      className="login"
    />
  </nav>
  <header
    className="header"
  >
    <div
      className=""
    >
      <div
        className="container content"
      >
        <a
          className="logo"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__img"
          />
        </a>
        <a
          className="smlogo"
          href="/"
          onClick={[Function]}
        >
          <span
            className="smlogo__img"
          />
        </a>
        <a
          className="cart ant-dropdown-trigger"
          href="/cart"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <span
            className="cart__icon"
          />
          <span
            className="cart__title"
          >
            购物车
          </span>
          <span
            className="cart__num"
          >
            0
          </span>
        </a>
        <a
          className="user"
          href="/user"
          onClick={[Function]}
        >
          <span
            className="user__icon"
          />
        </a>
        <div
          className="search"
        >
          <div
            className="search__main"
          >
            <div
              className="search__wrap"
            >
              <div
                className="search__hide"
              >
                &gt;&gt;
              </div>
              <div
                className="search__prefix"
              />
              <input
                className="search__input"
                onChange={[Function]}
                placeholder="搜索"
                type="text"
                value=""
              />
            </div>
            <div
              className="search__btn"
            >
              <div
                className="search__icon"
              />
              <div
                className="search__text"
                onClick={[Function]}
              >
                搜索
              </div>
            </div>
          </div>
        </div>
        <div
          className="loading"
        >
          <div
            className="ant-spin ant-spin-spinning"
          >
            <span
              aria-label="loading"
              className="anticon anticon-loading ant-spin-dot"
              role="img"
              style={
                Object {
                  "fontSize": 24,
                }
              }
            >
              <svg
                aria-hidden="true"
                className="anticon-spin"
                data-icon="loading"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="0 0 1024 1024"
                width="1em"
              >
                <path
                  d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
                />
              </svg>
            </span>
          </div>
        </div>
      </div>
    </div>
  </header>
  <div
    className="container content"
  >
    <div
      className="loading"
    >
      <div
        className="ant-spin ant-spin-spinning"
      >
        <span
          aria-label="loading"
          className="anticon anticon-loading ant-spin-dot"
          role="img"
          style={
            Object {
              "fontSize": 24,
            }
          }
        >
          <svg
            aria-hidden="true"
            className="anticon-spin"
            data-icon="loading"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 1024 1024"
            width="1em"
          >
            <path
              d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      className="order"
    >
      <div
        className="header"
      >
        <div
          className="header__product"
        >
          商品信息
        </div>
        <div
          className="header__price"
        >
          单价
        </div>
        <div
          className="header__quantity"
        >
          数量
        </div>
        <div
          className="header__sum"
        >
          小计
        </div>
        <div
          className="header__final"
        >
          实付
        </div>
      </div>
      <div
        className="body"
      >
        <div
          className="cart"
        >
          <div
            className="cart__item item"
          >
            <img
              alt="product"
              className="image"
              src="a"
            />
            <div
              className="body"
            >
              <div
                className="title"
              />
              <div
                className="specs"
              >
                <div
                  className="spec"
                >
                  规格1
                </div>
              </div>
            </div>
            <div
              className="price"
            >
              ¥
              179
            </div>
            <div
              className="quantity"
            >
              1
            </div>
            <div
              className="totalPrice"
            >
              ¥
              179
            </div>
            <div
              className="final"
            >
              ¥
              179
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="footer"
    >
      <div
        className="footer__row"
      >
        <div
          className="invoice"
        >
          <div
            className="invoice__title"
          >
            发票信息：
          </div>
          <div
            className="invoice__content"
          >
            <div
              className="invoice__check checkbox checkbox_none"
              type="checkbox"
            />
            <span
              className="invoice__text"
            >
              我要开发票
            </span>
          </div>
        </div>
        <div
          className="info"
        >
          <div
            className="info__row"
          >
            <div
              className="info__title"
            >
              商品合计 :
            </div>
            <div
              className="info__text"
            >
              ¥
              179
            </div>
          </div>
          <div
            className="info__row"
          >
            <div
              className="info__title"
            >
              运费 :
            </div>
            <div
              className="info__text"
            >
              ¥
              0
            </div>
          </div>
          <div
            className="info__row"
          >
            <div
              className="info__title"
            >
              活动优惠 :
            </div>
            <div
              className="info__text"
            >
              ¥
              0
            </div>
          </div>
        </div>
      </div>
      <div
        className="footer__row"
      >
        <div
          className="control"
        >
          <div
            className="sum"
          >
            <div
              className="sum__title"
            >
              应付总额 :
            </div>
            <div
              className="sum__text"
            >
              $
              179
            </div>
          </div>
          <div
            className="buy button"
            onClick={[Function]}
          >
            去付款
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="footer"
  >
    <div
      className="intro"
    >
      <div
        className="container content"
      >
        <div
          className="intro__col"
        >
          <div
            className="intro__title"
          >
            客户服务
          </div>
          <div
            className="intro__content"
          >
            <div
              className="buttons"
            >
              <div
                className="button"
              >
                <div
                  className="button__service"
                />
                <div
                  className="button__title"
                >
                  在线客服
                </div>
              </div>
              <div
                className="button"
              >
                <div
                  className="button__feedback"
                />
                <div
                  className="button__title"
                >
                  用户反馈
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="intro__col"
        >
          <div
            className="intro__title"
          >
            何为严选
          </div>
          <div
            className="intro__content"
          >
            <div
              className="text"
            >
              网易原创生活类电商品牌，秉承网易一贯的严谨态度，我们深入世界各地，从源头全程严格把控商品生产环节，力求帮消费者甄选到优质的商品
            </div>
            <div
              className="follow"
            >
              <div>
                关注我们 :
              </div>
              <div
                className="follow__qq"
              />
              <div
                className="follow__weixin"
              />
              <div
                className="follow__weibo"
              />
            </div>
          </div>
        </div>
        <div
          className="intro__col"
        >
          <div
            className="intro__title"
          >
            扫码下载严选APP
          </div>
          <div
            className="intro__content"
          >
            <div
              className="qr"
            >
              <img
                alt="qrcode"
                className="qr__image"
                src="qr.png"
              />
              <div
                className="qr__title"
              >
                下载领大额新人红包
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="link"
    >
      <div
        className="container content"
      >
        <div
          className="ensure"
        >
          <div
            className="ensure__item"
          >
            <div
              className="ensure__yi"
            />
            <div
              className="ensure__title"
            >
              网易自营电商
            </div>
          </div>
          <div
            className="ensure__item"
          >
            <div
              className="ensure__tui"
            />
            <div
              className="ensure__title"
            >
              30天无忧退换货
            </div>
          </div>
          <div
            className="ensure__item"
          >
            <div
              className="ensure__bao"
            />
            <div
              className="ensure__title"
            >
              满88元免邮费
            </div>
          </div>
          <div
            className="ensure__item"
          >
            <div
              className="ensure__ping"
            />
            <div
              className="ensure__title"
            >
              品质保证
            </div>
          </div>
        </div>
        <ul
          className="list"
        >
          <li
            className="list__item"
          >
            关于我们
          </li>
          <li
            className="list__item"
          >
            帮助中心
          </li>
          <li
            className="list__item"
          >
            售后服务
          </li>
          <li
            className="list__item"
          >
            配送与验收
          </li>
          <li
            className="list__item"
          >
            商务合作
          </li>
          <li
            className="list__item"
          >
            企业采购
          </li>
          <li
            className="list__item"
          >
            开放平台
          </li>
          <li
            className="list__item"
          >
            搜索推荐
          </li>
          <li
            className="list__item"
          >
            友情链接
          </li>
          <li
            className="list__item"
          >
            廉正举报
          </li>
        </ul>
        <ul
          className="certs"
        >
          <li
            className="certs__item"
          >
            食品经营许可证：JY13301080111719
          </li>
          <li
            className="certs__item"
          >
            出版物经营许可证：新出发滨字第0132号
          </li>
          <li
            className="certs__item"
          >
            妙得ICP证号：ICP 证浙B2-20160106
          </li>
          <li
            className="certs__item"
          >
            浙杭食药监械经营备20171029号
          </li>
          <li
            className="certs__item"
          >
            营业执照
          </li>
          <li
            className="certs__item"
          >
            网易公司版权所有 © 1997-2020
          </li>
        </ul>
        <ul
          className="certs"
        >
          <li
            className="certs__item"
          >
            无线电发射设备销售备案：42201919712647
          </li>
          <li
            className="certs__item"
          >
            (浙杭）网械企备字[2019]第00119号
          </li>
          <li
            className="certs__item"
          >
            （粤）网械平台备字（2019）第00004号
          </li>
          <li
            className="certs__item"
          >
            单用途商业预付卡备案证：330100AAC0024
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
